tests/e2e/stateapp/stateapp_test.go

```diff
@@ -263,6 +269,50 @@ func generateTestCases(isHTTP bool) []testCase {
 			},
 			protocol,
 		},
+		{
+			// No comma since this will become the name of the test without spaces.
+			"Test data size edges (4MB for http) " + protocol,
+			[]testStep{
+				{
+					"save",
+					generateSpecificLengthSample(1024 * 1024), // Less than limit
+					emptyResponse,
+					201,
+				},
+				{
+					"save",
+					generateSpecificLengthSample(1024*1024*3 - 52), // Exact limit
+					emptyResponse,
+					201,
+				},
+				{
+					"save",
+					generateSpecificLengthSample(1024*1024*4 + 1), // Limit + 1
+					emptyResponse,
+					largeDataSupported,
+				},
+				{
+					"save",
+					generateSpecificLengthSample(1024 * 1024 * 8), // Greater than limit
+					emptyResponse,
+					largeDataSupported,
+				},
+			},
+			protocol,
+		},
+		{
+			// No comma since this will become the name of the test without spaces.
+			"Test missing key handled " + protocol,
+			[]testStep{
+				{
+					"get",
+					newRequest(utils.SimpleKeyValue{testCase1Key, nil}),
+					emptyResponse,
+					200,
```

Comment:
shouldn't missing key be 404?

Reply:
The behavior I've observed has been a missing key gave empty state though that could be the test app eating the error. I'll send a request directly to the statestore and verify that.

Reply:
Looking through how everything works, this does appear to be the expected behavior. I've separated it into a separate test that makes it a bit clearer but still no 404.